src/borg/repository.py

```diff
@@ -344,6 +348,28 @@ def write_index(self):
             os.unlink(os.path.join(self.path, name))
         self.index = None
 
+    def check_free_space(self):
+        """Pre-commit check for sufficient free space to actually perform the commit."""
+        # Keep four times the (compact - not necessarily the case) index size free
```

Comment:
Yeah, I guess it'd be better to query the current hashtable size *and* apply a 4× factor here, instead of living with the 33-300 % variance in hashtable size vs. len(hashtable).